server/sonar-server/src/test/java/org/sonar/server/serverid/ws/ShowActionTest.java

```diff
@@ -0,0 +1,211 @@
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.api.utils.System2;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbTester;
+import org.sonar.db.property.PropertyDbTester;
+import org.sonar.server.exceptions.ForbiddenException;
+import org.sonar.server.platform.ServerIdGenerator;
+import org.sonar.server.tester.UserSessionRule;
+import org.sonar.server.ws.TestRequest;
+import org.sonar.server.ws.WsActionTester;
+import org.sonar.test.JsonAssert;
+import org.sonarqube.ws.MediaTypes;
+import org.sonarqube.ws.ServerId.ShowWsResponse;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.sonar.core.permission.GlobalPermissions.DASHBOARD_SHARING;
+import static org.sonar.core.permission.GlobalPermissions.SYSTEM_ADMIN;
+import static org.sonar.db.property.PropertyTesting.newGlobalPropertyDto;
+import static org.sonarqube.ws.MediaTypes.JSON;
+
+public class ShowActionTest {
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Rule
+  public UserSessionRule userSession = UserSessionRule.standalone();
+
+  @Rule
+  public DbTester db = DbTester.create(System2.INSTANCE);
+
+  DbClient dbClient = db.getDbClient();
+  PropertyDbTester propertyDb = new PropertyDbTester(db);
+  ServerIdGenerator generator = mock(ServerIdGenerator.class);
+
+  WsActionTester ws = new WsActionTester(new ShowAction(userSession, generator, dbClient));
+
+  @Test
+  public void return_server_id_info() throws Exception {
+    setUserAsSystemAdmin();
+    when(generator.generate("home", "127.0.1")).thenReturn("server_id");
+    mockAvailableIpAdresses("192.168.1.1", "127.0.1");
+    insertConfiguration("server_id", "home", "127.0.1");
+
+    ShowWsResponse response = executeRequest();
+
+    assertThat(response.getServerId()).isEqualTo("server_id");
+    assertThat(response.getOrganization()).isEqualTo("home");
+    assertThat(response.getIp()).isEqualTo("127.0.1");
+    assertThat(response.getValidIpAdressesList()).containsOnly("192.168.1.1", "127.0.1");
+    assertThat(response.hasInvalidServerId()).isFalse();
+  }
+
+  @Test
+  public void return_invalid_server_id() throws Exception {
+    setUserAsSystemAdmin();
+    when(generator.generate("home", "127.0.1")).thenReturn("server_id");
+    insertConfiguration("invalid", null, null);
+
+    ShowWsResponse response = executeRequest();
+
+    assertThat(response.hasInvalidServerId()).isTrue();
+    assertThat(response.getServerId()).isEqualTo("invalid");
+    assertThat(response.hasOrganization()).isFalse();
+    assertThat(response.hasIp()).isFalse();
+    assertThat(response.getValidIpAdressesList()).isEmpty();
+  }
+
+  @Test
+  public void return_nothing_when_no_settings_and_no_available_ips() throws Exception {
+    setUserAsSystemAdmin();
```

Comment:
suggestion: does it mean an empty json? maybe it would be better in this case to test the string output